File: src/main/java/HasUpper.java
import java.util.Scanner;

/**
 * @author pinker on 2018/1/9
 */
public class HasUpper {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String name;
        for (int i = 0; i < 5; i++) {
            name = in.nextLine();
            System.out.println(name);
            boolean isUpper = exists(name, Character::isUpperCase);
            System.out.println(isUpper);
        }
    }

    public static boolean exists(String str, CharacterProperty property) {
        boolean result = false;
        char[] arr = str.toCharArray();
        for (char ch : arr) {
            if (property.hasProperty(ch)) {
                result = true;
                break;
            }
        }
        return result;
    }
}

@FunctionalInterface
interface CharacterProperty {
    boolean hasProperty(char ch);
}

File: src/main/scala/scala_programing_book/HasUpper.scala
package scala_programing_book

/**
  * 此类的功能主要是判断一个String是否含有大小写,主要练习了scala中的高级特性,如果Java实现的
  * 话需要对每个字符遍历
  *
  * @author pinker on 2018/1/9
  */
object HasUpper extends App {
  val out = Console.in
  for (i <- 1 to 5) {
    val name = out.readLine()
    println(name.exists(_.isUpper).formatted("%s"))
  }
  out.close()
}
